sdk/keyvault/Azure.Security.KeyVault.Secrets/Readme.md

```diff
@@ -0,0 +1,255 @@
+
+```PowerShell
+Install-Package Azure.Security.KeyVault.Secrets -Version 1.0.0-preview.1
+```
+
+### Prerequisites
+* An [Azure subscription][azure_sub].
+* An existing Key Vault. If you need to create a Key Vault, you can use the Azure Portal or [Azure CLI][azure_cli].
+
+If you use the Azure CLI, replace `<your-resource-group-name>` and `<your-key-vault-name>` with your own, unique names:
+
+```PowerShell
+az keyvault create --resource-group <your-resource-group-name> --name <your-key-vault-name>
+```
+
+### Authenticate the client
+In order to interact with the Key Vault service, you'll need to create an instance of the [SecretClient][secret_client_class] class. You would need a **vault url** and **client secret credentials (client id, client secret, tenant id)** to instantiate a client object. 
+
+Client secret credential way of authentication is being used in this getting started section but you can find more ways to authenticate with [Azure identity][azure_identity].
+
+ #### Create/Get credentials
+Use the [Azure CLI][azure_cli] snippet below to create/get client secret credentials.
+
+ * Create a service principal and configure its access to Azure resources:
+    ```PowerShell
+    az ad sp create-for-rbac -n <your-application-name> --skip-assignment
+    ```
+    Output:
+    ```json
+    {
+        "appId": "generated-app-ID",
+        "displayName": "dummy-app-name",
+        "name": "http://dummy-app-name",
+        "password": "random-password",
+        "tenant": "tenant-ID"
+    }
+    ```
+* Use the above returned credentials information to set **AZURE_CLIENT_ID**(appId), **AZURE_CLIENT_SECRET**(password) and **AZURE_TENANT_ID**(tenant) environment variables. The following example shows a way to do this in Powershell:
+```cmd
+$Env:AZURE_CLIENT_ID="generated-app-ID"
+$Env:AZURE_CLIENT_SECRET="random-password"
+$Env:AZURE_TENANT_ID="tenant-ID"
+```
+
+* Grant the above mentioned application authorization to perform secret operations on the key vault:
+    ```PowerShell
+    az keyvault set-policy --name <your-key-vault-name> --spn $AZURE_CLIENT_ID --secret-permissions backup delete get list set
+    ```
+    > --secret-permissions:
+    > Accepted values: backup, delete, get, list, purge, recover, restore, set
+
+* Use the above mentioned Key Vault name to retreive details of your Vault which also contains your Key Vault URL:
+    ```PowerShell
+    az keyvault show --name <your-key-vault-name> 
+    ```
+
+#### Create Secret client
+Once you've populated the **AZURE_CLIENT_ID**, **AZURE_CLIENT_SECRET** and **AZURE_TENANT_ID** environment variables and replaced **your-vault-url** with the above returned URI, you can create the [SecretClient][secret_client_class]:
+
+```c#
+using Azure.Identity;
+using Azure.Security.KeyVault.Secrets;
+
+# Create a new secret client using the default credential from Azure.Identity
+var client = new SecretClient(vaultUri: <your-vault-url>, credential: new SystemCredential())
```

Comment:
It might be worth adding a comment explaining that SystemCredential picks up the env vars.

Reply:
Added